### PR TITLE
Dev branch

### LINKED ISSUE #2: Use different function for searching music and different for downloading

### PR DESCRIPTION
**Pull request by lunchspider**
using pytube instead of youtube_dl

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -35,15 +35,20 @@ class SongList:
     songs = []
     current_song_number : int
     def __init__(self, 
+            searchstr = None,
             song = None, 
             playlist = None, 
             shuffle = False, 
             repeatqueue = False,
-            repeatone = False):
+            repeatone = False,
```
Comment (Happy-Kunal) [inline src/main.py]: Is Running now ? you added a extra comma at the end of last argument of function

File: src/main.py [hunk 2]
```diff
@@ -35,15 +35,20 @@ class SongList:
...
+    ):
         if song: 
             self.songs.append(get_song_info(song))
         if playlist:
             self.songs.append(get_playlist_info(playlist))
+        if searchstr:
+            self.searchObject = youtubesearchpython.VideosSearch(music_choice,
```
Comment (Happy-Kunal) [inline src/main.py]: you might wanted to use searchstr instead of music_choice

**Review by Happy-Kunal — CHANGES_REQUESTED** [review]
Check the suggested reviews .

**Comment by lunchspider** [comment]
Done now

File: src/main.py [hunk 3]
```diff
@@ -60,6 +65,15 @@ def get_next_song(self) -> dict:
         if self.repeatqueue :
             if self.current_song_number == len(self.songs):
                 self.current_song_number = 0
+        if self.searchObject:
```
Comment (Happy-Kunal) [inline src/main.py:68]: Bhai ye error nahi dega aagr searchstr diya hi nahi hoga to ?

**Review by Happy-Kunal — CHANGES_REQUESTED** [review]
Bhai ek error aur hai😅😂

File: src/main.py (hunk 3 above)
Comment (lunchspider) [reply src/main.py:68]: ho gaya sahi ab

**Review by Happy-Kunal — APPROVED** [review]

**Pull request merged**